extra/src/main/java/aima/extra/probability/BigDecimalProbabilityNumber.java

```diff
@@ -0,0 +1,381 @@
+	 * 
+	 * @param that
+	 *            the BigDecimalProbabilityNumber that is to be subtracted from
+	 *            this BigDecimalProbabilityNumber.
+	 * 
+	 * @return a new BigDecimalProbabilityNumber that is the result of
+	 *         subtraction.
+	 */
+	@Override
+	public ProbabilityNumber subtract(ProbabilityNumber that) {
+		BigDecimalProbabilityNumber subtrahend = toInternalType(that);
+		int minPrecision = getMinPrecision(this.precisionMathContext.getPrecision(),
+				subtrahend.getMathContext().getPrecision());
+		MathContext resultMathContext = new MathContext(minPrecision, DEFAULT_PRECISION_ROUNDING_MODE);
+		return new BigDecimalProbabilityNumber(value.subtract(subtrahend.getValue(), resultMathContext),
+				resultMathContext);
+	}
+
+	/**
+	 * Multiply a BigDecimalProbabilityNumber with this
+	 * BigDecimalProbabilityNumber and return a new BigDecimalProbabilityNumber.
+	 * 
+	 * @param that
+	 *            the BigDecimalProbabilityNumber that is to be multiplied to
+	 *            this BigDecimalProbabilityNumber.
+	 * 
+	 * @result a new BigDecimalProbabilityNumber that is the result of
+	 *         multiplication.
+	 */
+	@Override
+	public ProbabilityNumber multiply(ProbabilityNumber that) {
+		BigDecimalProbabilityNumber multiplier = toInternalType(that);
+		int minPrecision = getMinPrecision(this.precisionMathContext.getPrecision(),
+				multiplier.getMathContext().getPrecision());
+		MathContext resultMathContext = new MathContext(minPrecision, DEFAULT_PRECISION_ROUNDING_MODE);
+		return new BigDecimalProbabilityNumber(value.multiply(multiplier.value, resultMathContext), resultMathContext);
+	}
+
+	/**
+	 * Divide a BigDecimalProbabilityNumber with this
+	 * BigDecimalProbabilityNumber and return a new BigDecimalProbabilityNumber.
+	 * 
+	 * @param that
+	 *            the BigDecimalProbabilityNumber that is the divisor of this
+	 *            BigDecimalProbabilityNumber.
+	 * 
+	 * @return a new BigDecimalProbabilityNumber that is the result of division.
+	 */
+	@Override
+	public ProbabilityNumber divide(ProbabilityNumber that) {
+		BigDecimalProbabilityNumber divisor = toInternalType(that);
+		if (divisor.isZero()) {
+			throw new IllegalArgumentException();
+		}
+		int minPrecision = getMinPrecision(this.precisionMathContext.getPrecision(), divisor.getMathContext().getPrecision());
+		MathContext resultMathContext = new MathContext(minPrecision, DEFAULT_PRECISION_ROUNDING_MODE);
+		return new BigDecimalProbabilityNumber(value.divide(divisor.value, resultMathContext), resultMathContext);
+	}
+
+	/**
+	 * Calculate the BigDecimalProbabilityNumber raised to an integer exponent.
+	 * 
+	 * @param exponent
+	 *            of integer type.
+	 * 
+	 * @result a new BigDecimalProbabilityNumber that is this
+	 *         BigDecimalProbabilityNumber raised to the exponent value.
+	 */
+	@Override
+	public ProbabilityNumber pow(int exponent) {
+		return new BigDecimalProbabilityNumber(value.pow(exponent, precisionMathContext));
```

Comment:
what happens if exponent > EXPONENT_MAX, you should handle this here and delegate to the BigInteger version.